extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/RouteBuildItem.java

```diff
@@ -192,17 +224,39 @@ public Builder displayOnNotFoundPage(String notFoundPageTitle) {
             return this;
         }
 
+        /**
+         * @deprecated Specify the path as part of defining the route
+         * @see #route(String)
+         * @see #routeFunction(String, Consumer)
+         */
+        @Deprecated
         public Builder displayOnNotFoundPage(String notFoundPageTitle, String notFoundPagePath) {
             this.displayOnNotFoundPage = true;
             this.notFoundPageTitle = notFoundPageTitle;
             this.notFoundPagePath = notFoundPagePath;
             return this;
         }
 
+        public Builder routeConfigKey(String attributeName) {
+            this.routeConfigKey = attributeName;
+            return this;
+        }
+
         public RouteBuildItem build() {
             return new RouteBuildItem(this, RouteType.APPLICATION_ROUTE, false);
         }
 
+        protected ConfiguredPathInfo getRouteConfigInfo() {
```

Comment:
Wondering if it's beneficial to pass a value for "absolute" or not into the Builder so that this method doesn't need to be overridden with the same content in the two children?

Reply:
It's a little more complicated than that because of where/how the absolutely resolved path is known/stored